frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.tsx

```diff
@@ -24,51 +28,63 @@ import {
 } from './parseUtils';
 import { PipelineTask, PipelineTaskRunStatus } from './pipelineTaskTypes';
 
-const idForTaskArtifact = (groupId: string | undefined, artifactId: string) =>
-  groupId ? `${groupId}-ARTIFACT-${artifactId}` : artifactId;
+const getArtifactPipelineTask = (
+  name: string,
+  artifactType: InputOutputArtifactType,
+): PipelineTask => ({
+  type: 'artifact',
+  name,
+  inputs: {
+    artifacts: [{ label: name, type: composeArtifactType(artifactType) }],
+  },
+});
+
+/**
+ * Get the artifact nodes without the inputs from a task node
+ */
+const getInputArtifacts = (
```

Comment:
This is important to solve [RHOAIENG-6671](https://issues.redhat.com/browse/RHOAIENG-6671), some artifact nodes are not from tasks, so they have no inputs, we need to find them from the input definitions.